### PR TITLE
fix(InventoryTable): ESSNTL-3505 - Only use columns already in the store

### PR DESCRIPTION
**Pull request by bastilian**
**Steps to reproduce:**

1.) Open the "Settings" -> "Remote Host Configuration" page.
2.) Click the top right "View log" link
3.) Go to the "Systems" tab (Make sure there are systems in the list, otherwise the error is not triggered)
4.) Close the modal.
5.) Enter `window.__scalprum__.scalprumOptions.cacheTimeout = 1;` into the console to force a lower module cache timeout.
6.) After waiting reopen the "View log" modal

On Prod/Stage this should throw errors and not load the inventory. With this PR the errors should not occur and the Inventory should load normally.

### CODE REVIEW

**Comment by gkarat** [comment]
Is there a concrete way of how to reproduce the bug?

**Comment by bastilian** [comment]
@gkarat I added steps to reproduce the issue.

**Review by gkarat — APPROVED** [review]
Well, it does looks OK to me, good job fixing the issue 👍🏼 And I couldn't reproduce the bug while running locally with this PR.

However, I put some questions to some parts of the code. But I see that they all concern the legacy code that was before this PR. So probably, if there is something that makes sense, can be addressed separately.

File: src/components/InventoryTable/hooks/useColumns.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                columnsProp
+            ], 'key');
+        } else if (!columnsProp && columnsRedux) {
+            return columnsRedux;
+        }  else {
+            return defaultColumnsFiltered;
+        }
+    }, [
+        showTags,
+        Array.isArray(disableDefaultColumns) ? disableDefaultColumns.join() : disableDefaultColumns,
```
[the remark this answers is not in the document]
Comment (gkarat) [reply src/components/InventoryTable/hooks/useColumns.js:38]: Ah, ok, in this case this looks legit

**Review by adonispuente — APPROVED** [review]
I was unable to reproduce the bug. This PR works as intended, LGTM

**Pull request merged**